src/Controller/ExportController.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace App\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+class ExportController extends AbstractController
+{
+    /**
+     * @Route("/students.json", name="export")
+     */
+    public function index()
+    {
+
+        $data = [
+            'nedarykpats' => [
+                'name' => 'Nedaryk pats',
+                'mentors' => [
+                    'Laurynas'
+                ],
+                'students' => [
+                    'Martyna',
+                    'Aurimas',
+                    'Vilius'
+                ]
+            ],
+            'savanoryste' => [
+                'name' => 'Car booking',
+                'mentors' => [
+                    'Tomas'
+                ],
+                'students' => [
+                    'Ignas',
+                    'Dovydas',
+                    'Darius'
+                ]
+            ],
+            'curlybrackets' => [
+                'name' => 'Maisto dalinimosi sistema',
+                'mentors' => [
+                    'Paulius'
+                ],
+                'students' => [
+                    'Roman',
+                    'Marijus',
+                    'Angelika'
+                ]
+            ],
+            'hobby' => [
+                'name' => 'Hobby',
+                'mentors' => [
+                    'Ieva'
+                ],
+                'students' => [
+                    'Miroslav',
+                    'Viktoras',
+                    'Lukas'
+                ]
+            ],
+            'hack<b>er</b>\'is po .mySubdomain &project=123' => [
+                'name' => '\' OR 1 -- DROP DATABASE',
+                'mentors' => [
+                    '<b>Ponas</b> Programišius'
+                ],
+                'students' => [
+                    'Aurelijus Banelis',
+                    '<b>Ir</b> jo \"geras\" draug\'as'
+                ]
+            ],
+        ];
+
+
+        $response = new Response(json_encode($data, JSON_PRETTY_PRINT));
```

Comment:
Dabar veikia, bet verta žinoti ir apie alternatyvą: `JsonResponse`

Plačiau:
 * https://symfony.com/doc/current/components/http_foundation.html#creating-a-json-response

Reply:
Sutinku, kad galima, bet tada gražinamas json'o view'sas nebus toks pat kaip ir pas jus. T.y. nebus "JSON_PRETTY_PRINT"

Reply:
Šioje vietoje jau perdariau kitu būdų

Reply:
`Response` objektas turi ir `header` parametrus. Pagal `Content-type` reikšmę naršyklė gali teisingiau interpretuoti, ar tai yra tekstas, JSON failas, paveiklsėlis ar pan.

Tavo atveju:
```php
$response = new BinaryFileResponse($file);
```
Yra analogiškas:
```php
$response = new BinaryFileResponse($file, $status = 200, array $headers = [])
```
Kai tuo tarpu `JsonResponse` dar prideda `Content-type` reikšmę:
https://github.com/symfony/http-foundation/blob/dba4ae85586fd75ce37a4429be772643d7d6bfe2/JsonResponse.php#L199

Todėl lyginant:
```bash
curl -v http://127.0.0.1:8000/students.json 2>&1 | grep Content
curl -v https://nfq190418.realus.website/students.json 2>&1 | grep Content
```
Atsakymai skiriasi:
![image](https://user-images.githubusercontent.com/1453957/57176369-e1c14900-6e5f-11e9-9da4-193375dc88a3.png)

Ir `JSON_PRETTY_PRINT` interpretuoja jau naršyklė, nes žino, kad čia yra `application/json`, o ne `text/plain`

Plačiau:
 * https://stackoverflow.com/questions/477816/what-is-the-correct-json-content-type

Reply:
Savo pavyzdyje aš tiesiog įdėjau failą į `public` katalogą ir pagal `nginx` konfiguraciją, jei failas egzistuoja – jį grąžina `nginx` (kuris uždeda visus `Content-type` ir pan.).
Jei rašai `/student` – tokio failo nėra, todėl užklausą nukreipia į `index.php` failą

Plačiau:
 * https://symfony.com/doc/current/setup/web_server_configuration.html#nginx

Reply:
P.S. dėl šito balo nemažinsiu (taisyti nebūtina) – tiesiog ateičiai verta suvokti :wink:

Reply:
Supratau, dėkui už pavyzdžius!